src/__test__/server.test.js

```diff
@@ -0,0 +1,196 @@
+        .then(Promise.reject)
+        .catch((response) => {
+          expect(response.status).toEqual(404);
+        });
+    });
+    test('GET - should respond with an array if no id specified.', () => {
+      return superagent.get(apiURL)
+        .then((response) => {
+          expect(Array.isArray(response.body)).toBeTruthy();
+        });
+    });
+  });
+
+  describe('DELETE /api/users', () => {
+    test('DELETE - should respond with 204 if item was successfully removed.', () => {
+      return createUserMock()
+        .then((user) => {
+          return superagent.delete(`${apiURL}/${user._id}`);
+        })
+        .then((response) => {
+          expect(response.status).toEqual(204);
+        });
+    });
+    test('DELETE - should respond with 404 if there is no user to be found', () => {
+      return superagent.delete(`${apiURL}/THisIsAnInvalidId`)
+        .then(Promise.reject)
+        .catch((response) => {
+          expect(response.status).toEqual(404);
+        });
+    });
+  });
+
+  describe('PUT /api/users', () => {
+    test('PUT - should update a user and return a 200 status code.  ', () => {
+      let userToUpdate = null;
+      return createUserMock()
+        .then((userMock) => {
+          userToUpdate = userMock;
+          return superagent.put(`${apiURL}/${userMock._id}`)
+            .send({ name: 'John Doe' });
+        })
+        .then((response) => {
+          expect(response.status).toEqual(200);
+          expect(response.body.name).toEqual('John Doe');
+          expect(response.body.location).toEqual(userToUpdate.location);
+          expect(response.body.bicepsRPM).toEqual(userToUpdate.bicepsRPM);
+          expect(response.body.tricepsRPM).toEqual(userToUpdate.tricepsRPM);
+          expect(response.body._id).toEqual(userToUpdate._id.toString());
+          // Mongo generated id's are objects, not strings/numbers thus toString() is required.
+        });
+    });
+    test('PUT - should respond with a 404 if user not found.', () => {
+      return createUserMock()
+        .then(() => {
+          return superagent.put(`${apiURL}/invalidID`)
+            .send({ name: 'John Doe' });
+        })
+        .catch((error) => {
+          expect(error.status).toEqual(404);
+        });
+    });
+    test('PUT - should respond with a 400 if invalid id.', () => {
+      return createUserMock()
+        .then((userMock) => {
+          return superagent.put(`${apiURL}/${userMock._id}`)
+            .send({ name: '' });
+        })
+        .catch((error) => {
+          expect(error.status).toEqual(400);
+        });
+    });
+    test('PUT - 409 due to duplicate name', () => {
+      const firstMock = () => {
+        return new User({
+          name: 'John Doe',
+          location: faker.address.city(),
+          bicepsRPM: faker.random.number(300),
+          tricepsRPM: faker.random.number(300),
+        }).save();
+      };
+      return createUserMock()
+        .then((userMock) => {
```

Comment:
Lots going on with this test that need some fixing. I'll give you a version of code to use, but I want you to inspect this test on your own time to realize what's going on. I want you console.log `firstMock.name` and see what you get. I want you to also read up on this MDN doc:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name